tests/functional/test_config_groups.py

```diff
@@ -110,51 +132,128 @@ def test_that_groups_not_allowed_to_intersect_in_cluster(self, sdk_client_fs, cl
             service_group.host_add(test_host_1)
         with allure.step("Create the second group for service and check that not allowed to add the first host to it"):
-            service_group_2 = service.group_config_create(name=self.SECOND_GROUP)
-            self.assert_that_host_add_is_unavailable(service_group_2, test_host_1)
-            self.assert_host_candidate_equal_expected(service_group_2.host_candidate(), 1)
+            service_group_2 = service.group_config_create(name=SECOND_GROUP)
+            assert_that_host_add_is_unavailable(service_group_2, test_host_1)
+            assert_host_candidate_equal_expected(service_group_2.host_candidate(), [SECOND_HOST])
 
     def test_that_groups_not_allowed_to_intersect_in_component(self, sdk_client_fs, cluster_with_components):
         """Test that groups are not allowed to intersect"""
 
         service, test_host_1, _ = cluster_with_components
         with allure.step("Create config group for component and add the first host"):
-            component_group = service.component(name=self.FIRST_COMPONENT_NAME).group_config_create(
-                name=self.FIRST_GROUP
-            )
+            component_group = service.component(name=FIRST_COMPONENT_NAME).group_config_create(name=FIRST_GROUP)
             component_group.host_add(test_host_1)
         with allure.step(
             "Create the second group for component and check that not allowed to add the first host to it"
         ):
-            component_group_2 = service.component(name=self.FIRST_COMPONENT_NAME).group_config_create(
-                name=self.SECOND_GROUP
-            )
-        self.assert_that_host_add_is_unavailable(component_group_2, test_host_1)
-        self.assert_host_candidate_equal_expected(component_group_2.host_candidate(), 1)
+            component_group_2 = service.component(name=FIRST_COMPONENT_NAME).group_config_create(name=SECOND_GROUP)
+        assert_that_host_add_is_unavailable(component_group_2, test_host_1)
+        assert_host_candidate_equal_expected(component_group_2.host_candidate(), [SECOND_HOST])
+
+
+class TestIncorrectHostInGroups:
+    @pytest.fixture()
+    def cluster_with_components_on_one_host(
+        self, create_two_hosts, cluster: Cluster, provider: Provider
+    ) -> Tuple[Service, Host, Host]:
+        """Add service, two hosts and create components to check config groups"""
+
+        service = cluster.service_add(name='test_service_1')
+        test_host_1, test_host_2 = create_two_hosts
+        cluster.host_add(test_host_1)
+        cluster.hostcomponent_set(
+            (test_host_1, service.component(name=FIRST_COMPONENT_NAME)),
+            (test_host_1, service.component(name=SECOND_COMPONENT_NAME)),
+        )
+        return service, test_host_1, test_host_2
+
+    def test_add_incorrect_host_to_provider_group(self, sdk_client_fs, provider_bundle, provider):
+        """Test exception rise when we try to add incorrect host to provider group"""
```

Comment:
Add to test description what is mean by "incorrect host"

Reply:
added to docstring in class